packages/tenant-process/src/services/tenantService.ts

```diff
@@ -1153,6 +1157,63 @@ export function tenantServiceBuilder(
       logger.info(`Retrieving Tenant with Selfcare Id ${selfcareId}`);
       return readModelService.getTenantBySelfcareId(selfcareId);
     },
+
+    async addCertifierId(
+      {
+        tenantId,
+        certifierId,
+        correlationId,
+      }: {
+        tenantId: TenantId;
+        certifierId: string;
+        correlationId: string;
+      },
+      logger: Logger
+    ): Promise<Tenant> {
+      logger.info(`Adding certifierId to Tenant ${tenantId}`);
+
+      const tenant = await retrieveTenant(tenantId, readModelService);
+
+      const certifierFeature = tenant.data.features.find((a) => a.certifierId);
+
+      if (certifierFeature) {
+        if (certifierId === certifierFeature.certifierId) {
+          throw certifierIdAlreadyExistsInTenant(certifierId, tenant.data.id);
+        }
+
+        const certifiedAttribute =
+          await readModelService.getOneCertifiedAttributeByCertifier({
+            certifierId: certifierFeature.certifierId,
+          });
+        if (certifiedAttribute) {
+          throw tenantIsAlreadyACertifier(
+            tenant.data.id,
+            certifierFeature.certifierId
+          );
+        }
+      }
+
+      const updatedTenant: Tenant = {
+        ...tenant.data,
+        features: [
+          ...tenant.data.features,
```

Comment:
we need to filter out features with type `PersistentCertifier`.
Previous checks pass if the tenant has already a `certifierId` different from the requested one, and that has no attributes